Purchases/Networking/Operations/PostReceiptDataOperation.swift

```diff
@@ -59,38 +57,8 @@ class PostReceiptDataOperation: CacheableNetworkOperation {
     }
 
     private func post(completion: @escaping () -> Void) {
-        let fetchToken = self.postData.receiptData.asFetchToken
-        var body: [String: Any] = [
-            "fetch_token": fetchToken,
-            "app_user_id": self.configuration.appUserID,
-            "is_restore": self.postData.isRestore,
-            "observer_mode": self.postData.observerMode
-        ]
-
-        if let productData = self.postData.productData {
-            do {
-                body += try productData.asDictionary()
```

Comment:
I wonder when this could actually throw

Reply:
If it failed to encode it:
```swift
func asDictionary() throws -> [String: Any] {
        let data = try JSONEncoder().encode(self)
        let result = try JSONSerialization.jsonObject(with: data, options: [])

        guard let result = result as? [String: Any] else {
            throw CodableError.unexpectedValue(type(of: result))
        }

        return result
    }
```

Reply:
right, I meant in practice, when would the productData not be convertible to a dictionary

Reply:
Oh in practice never.